### DISCUSSION TITLE
Speech to Text / Voice control for Chat

### DISCUSSION THREAD

**Discussion by gamesbyjames** [google/mesop#576] [Q&A] [answered]
Anyone have any recommendations for adding voice control for the Chat component? Thanks.

**Comment by richard-to** [comment, marked as answer]
We currently don't have a voice control. But I think you can create a voice control using the web components feature.

You can also use this closed Pull Request as a guide -> https://github.com/google/mesop/pull/290. This example is using the core component approach, so you'd need to modify it for web components structure (https://google.github.io/mesop/web_components/)

Once you have the speech to text / voice control created, you can copy https://github.com/google/mesop/blob/main/mesop/labs/chat.py to your own chat.py. From there you can add your voice control to your custom chat.py file.

Overall, it's not trivial, but definitely possible.
